### PR TITLE
Change the SqlParquetReader to wrap the actual ParquetReader and add …

### PR DESCRIPTION
**Pull request by guillaumer63**
…feature to read columns from file

### CODE REVIEW

File: src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java [hunk 1]
```diff
@@ -16,26 +16,53 @@
...
-    return new SQLParquetReader.Builder(inputFile);
+
+    ParquetReader.Builder<Record> recordParquetReader = new SQLParquetReaderWrapper.Builder(inputFile);
+
+    this.parquetReader = recordParquetReader.build();
+    this.schema = this.initFileSchema(inputFile);
+  }
+
+  private MessageType initFileSchema(InputFile inputFile) throws IOException {
+    ParquetFileReader parquetFileReader = ParquetFileReader.open(inputFile);
```
Comment (anthoRx) [inline src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java:48]: ParquetFileReader should be close when MessageType is retrieved

File: src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java [hunk 2]
```diff
@@ -16,26 +16,53 @@
...
+
+  public MessageType getSchema() {
+    return schema;
+  }
+
+  public Record read() throws IOException {
+    return parquetReader.read();
+  }
+
+  public List<String> getColumns() {
```
Comment (anthoRx) [inline src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java:60]: getFieldsNames is a better name in my opinion, we don't speak about column in Parquet.

File: src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java (hunk 1 above)
Comment (guillaumer63) [reply src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java:48]: Good point, will fix it. 👍

File: src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java (hunk 2 above)
Comment (guillaumer63) [reply src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java:60]: Ok, will change it, thanks.

File: src/main/java/io/github/anthorx/parquet/sql/write/JDBCWriter.java [hunk 3]
```diff
@@ -81,4 +83,20 @@ public void write() throws IOException, SQLException {
       }
     }
   }
+
+  private List<RecordField> extractFieldValues(List<String> fieldsNames, Record record) {
+    return fieldsNames
+        .stream()
+        .reduce(new ArrayList<>(), (currentValues, currentFieldName) -> {
+          RecordField result = getNextField(currentFieldName, record);
```
Comment (anthoRx) [inline src/main/java/io/github/anthorx/parquet/sql/write/JDBCWriter.java:91]: I think you can avoid using getNextField function which introduce boilerplate code by using orElse directly :
```
RecordField result = record
             .getField(currentFieldName)
             .orElse(new RecordField<>(currentFieldName, null))
```

Comment (guillaumer63) [reply src/main/java/io/github/anthorx/parquet/sql/write/JDBCWriter.java]: Yeah, refactored the method and don't see this simplification :/

**Pull request merged**